lib/buffer.js

```diff
@@ -933,8 +940,12 @@ Buffer.prototype.readUInt32BE = function(offset, noAssert) {
 
 
 Buffer.prototype.readIntLE = function(offset, byteLength, noAssert) {
+  if (!noAssert)
+    checkByteLength(byteLength);
```

Comment:
Probably best moving `checkByteLength` and  `checkOffset` in the same branch. Also not sure if the assertion should happen before or after `offset` and `byteLength` get converted into numbers, I am inclined to checking it before the conversion.